build_tools/update_environments_and_lock_files.py

```diff
@@ -192,6 +192,12 @@ def remove_from(alist, to_remove):
         ),
         "package_constraints": {
             "python": "3.9",
+            # TODO: remove the following pin when we have a solution since the current
+            # environment defeat the purpose of the test (e.g. `pylastest`).
+            # pandas 2.0 depends on tzdata as well as python installed via conda
+            # We currently have a mix of conda/pip and run into the following issue:
+            # https://github.com/conda/conda-lock/issues/179
+            "pandas": "1.5.3",
```

Comment:
what happens if we get it from conda-forge instead of pypi?

Reply:
Then we defeat the purpose of installing everything via `pip` as intended.
Here, I chose to keep what is currently happening on the CI.

Reply:
hmm. so here's my issue. One can pip install `pandas=2.0` in a conda-forge environment w/o an issue. Loading `tzdata` in a `python` process will then load the one installed by `pip`. So from the user's perspective, they can use the latest pandas, and scikit-learn, in a conda-forge env. But here in the CI now we're not testing that.

<details>

```shell
~ ⌚ 11:26:06
$ mmamba create -n delete-me python=3.11       

                                           __
          __  ______ ___  ____ _____ ___  / /_  ____ _
         / / / / __ `__ \/ __ `/ __ `__ \/ __ \/ __ `/
        / /_/ / / / / / / /_/ / / / / / / /_/ / /_/ /
       / .___/_/ /_/ /_/\__,_/_/ /_/ /_/_.___/\__,_/
      /_/

conda-forge/noarch                                  11.9MB @   3.2MB/s  3.8s
conda-forge/linux-64                                30.9MB @   3.4MB/s  9.4s

Transaction

  Prefix: /home/adrin/miniforge3/envs/delete-me

  Updating specs:

   - python=3.11


  Package               Version  Build               Channel                    Size
──────────────────────────────────────────────────────────────────────────────────────
  Install:
──────────────────────────────────────────────────────────────────────────────────────

  + _libgcc_mutex           0.1  conda_forge         conda-forge/linux-64     Cached
  + _openmp_mutex           4.5  2_gnu               conda-forge/linux-64     Cached
  + bzip2                 1.0.8  h7f98852_4          conda-forge/linux-64     Cached
  + ca-certificates   2022.12.7  ha878542_0          conda-forge/linux-64     Cached
  + ld_impl_linux-64       2.40  h41732ed_0          conda-forge/linux-64     Cached
  + libexpat              2.5.0  hcb278e6_1          conda-forge/linux-64       78kB
  + libffi                3.4.2  h7f98852_5          conda-forge/linux-64     Cached
  + libgcc-ng            12.2.0  h65d4601_19         conda-forge/linux-64     Cached
  + libgomp              12.2.0  h65d4601_19         conda-forge/linux-64     Cached
  + libnsl                2.0.0  h7f98852_0          conda-forge/linux-64     Cached
  + libsqlite            3.40.0  h753d276_0          conda-forge/linux-64     Cached
  + libuuid              2.38.1  h0b41bf4_0          conda-forge/linux-64     Cached
  + libzlib              1.2.13  h166bdaf_4          conda-forge/linux-64     Cached
  + ncurses                 6.3  h27087fc_1          conda-forge/linux-64     Cached
  + openssl               3.1.0  h0b41bf4_0          conda-forge/linux-64     Cached
  + pip                    23.1  pyhd8ed1ab_0        conda-forge/noarch          1MB
  + python               3.11.3  h2755cc3_0_cpython  conda-forge/linux-64       31MB
  + readline                8.2  h8228510_1          conda-forge/linux-64     Cached
  + setuptools           67.6.1  pyhd8ed1ab_0        conda-forge/noarch       Cached
  + tk                   8.6.12  h27826a3_0          conda-forge/linux-64     Cached
  + tzdata                2023c  h71feb2d_0          conda-forge/noarch        118kB
  + wheel                0.40.0  pyhd8ed1ab_0        conda-forge/noarch       Cached
  + xz                    5.2.6  h166bdaf_0          conda-forge/linux-64     Cached

  Summary:

  Install: 23 packages

  Total download: 33MB

──────────────────────────────────────────────────────────────────────────────────────


Confirm changes: [Y/n] 

Transaction starting
tzdata                                             117.6kB @ 352.0kB/s  0.3s
libexpat                                            78.0kB @ 220.3kB/s  0.4s
pip                                                  1.4MB @   2.8MB/s  0.5s
python                                              31.1MB @  14.4MB/s  2.2s
Linking _libgcc_mutex-0.1-conda_forge
Linking ld_impl_linux-64-2.40-h41732ed_0
Linking ca-certificates-2022.12.7-ha878542_0
Linking libgomp-12.2.0-h65d4601_19
Linking _openmp_mutex-4.5-2_gnu
Linking libgcc-ng-12.2.0-h65d4601_19
Linking libffi-3.4.2-h7f98852_5
Linking bzip2-1.0.8-h7f98852_4
Linking ncurses-6.3-h27087fc_1
Linking openssl-3.1.0-h0b41bf4_0
Linking libuuid-2.38.1-h0b41bf4_0
Linking libexpat-2.5.0-hcb278e6_1
Linking xz-5.2.6-h166bdaf_0
Linking libzlib-1.2.13-h166bdaf_4
Linking libnsl-2.0.0-h7f98852_0
Linking readline-8.2-h8228510_1
Linking tk-8.6.12-h27826a3_0
Linking libsqlite-3.40.0-h753d276_0
Linking tzdata-2023c-h71feb2d_0
Linking python-3.11.3-h2755cc3_0_cpython
Linking wheel-0.40.0-pyhd8ed1ab_0
Linking setuptools-67.6.1-pyhd8ed1ab_0
Linking pip-23.1-pyhd8ed1ab_0
Transaction finished

~ ⌚ 12:22:24
$ mmamba activate delete-me
(delete-me) 
~ ⌚ 12:22:34
$ pip install pandas
Collecting pandas
  Downloading pandas-2.0.0-cp311-cp311-manylinux_2_17_x86_64.manylinux2014_x86_64.whl (12.2 MB)
     ━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━ 12.2/12.2 MB 16.0 MB/s eta 0:00:00
Collecting python-dateutil>=2.8.2 (from pandas)
  Using cached python_dateutil-2.8.2-py2.py3-none-any.whl (247 kB)
Collecting pytz>=2020.1 (from pandas)
  Downloading pytz-2023.3-py2.py3-none-any.whl (502 kB)
     ━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━ 502.3/502.3 kB 16.0 MB/s eta 0:00:00
Collecting tzdata>=2022.1 (from pandas)
  Downloading tzdata-2023.3-py2.py3-none-any.whl (341 kB)
     ━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━ 341.8/341.8 kB 18.3 MB/s eta 0:00:00
Collecting numpy>=1.21.0 (from pandas)
  Downloading numpy-1.24.2-cp311-cp311-manylinux_2_17_x86_64.manylinux2014_x86_64.whl (17.3 MB)
     ━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━ 17.3/17.3 MB 16.3 MB/s eta 0:00:00
Collecting six>=1.5 (from python-dateutil>=2.8.2->pandas)
  Using cached six-1.16.0-py2.py3-none-any.whl (11 kB)
Installing collected packages: pytz, tzdata, six, numpy, python-dateutil, pandas
Successfully installed numpy-1.24.2 pandas-2.0.0 python-dateutil-2.8.2 pytz-2023.3 six-1.16.0 tzdata-2023.3
(delete-me) 
~ ⌚ 12:22:47
$ python
Python 3.11.3 | packaged by conda-forge | (main, Apr  6 2023, 08:57:19) [GCC 11.3.0] on linux
Type "help", "copyright", "credits" or "license" for more information.
>>> import tzdata
>>> tzdata.__version__
'2023.3'
>>> 
```

</details>

Reply:
We are testing pandas development version in CI installed from pip is that good enough?

Reply:
+1 for merging this as it is, and if the conda-lock problem does not get resolved in the next few months, open a follow-up PR to add the (unpinned) pandas optional dependency to one of the `pylatest_conda_forge_*` configurations.